indico/modules/designer/client/js/index.js

```diff
@@ -424,8 +430,63 @@ import {$T} from 'indico/utils/i18n';
           selectedItem.y = unzoom(y > 0 ? y : 0);
         }
       },
+      back: function() {
+        if (div) {
+          var zIndex = getZIndex(selectedItem, false);
+          div.css('zIndex', zIndex);
+          selectedItem.zIndex = zIndex;
+        }
+      },
+      front: function() {
+        if (div) {
+          var zIndex = getZIndex(selectedItem, true);
+          div.css('zIndex', zIndex);
+          selectedItem.zIndex = zIndex;
+        }
+      },
     }[direction]());
   }
+  function itemsOverlap(item1, item2) {
+    const left1 = item1.x;
+    const top1 = item1.y;
+    const right1 = item1.x + item1.width;
+    const bottom1 = item1.y + (item1.height || 22);
+    const left2 = item2.x;
+    const top2 = item2.y;
+    const right2 = item2.x + item2.width;
+    const bottom2 = item2.y + (item2.height || 22);
+
+    if (left1 >= right2 || left2 >= right1) {
+      return false;
+    }
+
+    if (top1 >= bottom2 || top2 >= bottom1) {
+      return false;
+    }
+    return true;
+  }
+
+  function getZIndex(selectedItem, moveToFront) {
+    let zIndex = selectedItem.zIndex;
+    for (const i in items) {
+      if (selectedItem.id === items[i].id) {
+        continue;
+      }
+      if (itemsOverlap(selectedItem, items[i])) {
+        if (moveToFront) {
+          zIndex = Math.max(items[i].zIndex, zIndex);
+        } else {
+          zIndex = Math.min(items[i].zIndex, zIndex);
+        }
+      }
+    }
+    const randomZIndexValue = Math.floor(Math.random() * 10) + 5;
```

Comment:
Why is this needed? Using randomness seems really strange here.

Reply:
I can change this to always increase/decrease by 1. it would work also